### PR TITLE
[FIX] OWMap: updating on removing/adding lat and long

### PR DESCRIPTION
**Pull request by robertcv**
##### Issue
If all potential lat/long attributes are removed at once and then re-added the map doesn't update to new data because effective variables are empty which means they are the same.

##### Description of changes
If no valid lat/long attribute selection exists the set `self.data = None`. This mimics the behavior of Scatter plot with no continues variables.

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: orangecontrib/geo/widgets/owmap.py [hunk 1]
```diff
@@ -167,9 +168,14 @@ def _add_controls(self):
...
+        self.Warning.no_lat_lon_vars.clear()
+        self._attr_lat, self._attr_lon = None, None
         if self.data is not None:
-            if not self.data.domain.has_continuous_attributes(True, True):
-                self.Warning.no_continuous_vars()
+            self._attr_lat, self._attr_lon = \
+                find_lat_lon(self.data, filter_hidden=True)
+            if self._attr_lat is None or self._attr_lon is None:
+                # we either find both or none
+                self.Warning.no_lat_lon_vars()
```
Comment (janezd) [inline orangecontrib/geo/widgets/owmap.py:178]: Same code, same comment as in #100. When setting `self.data` to `None`, I think the widget should also clean everything else, as a matter of hygiene. Hence, I think you should set `attr_lan` and `attr_lon` to `None`.

Also, consider upgrading `no_lat_lon_vars` from a warning to an error.

**Comment by janezd** [comment]
It does the job, so I'm merging. There is something interesting, though: connect File -> Select Columns -> Geo. Plot data. Use "Select Rows" to remove "Long". Geo will plot Lat vs. Lat. I suppose this is not the expected behaviour.

Furthermore, re-add Long and the widget will still plot Lat vs. Lat.

I suppose this should be fixed in `find_lat_lon`, but it's not critical.

**Comment by robertcv** [comment]
Your example works as follows: after you remove Long the find_lat_lon finds the Lat attr by name and because it doesn't find Long by name it will take the first continuous variable which values are inside [-180, 180]. So Lat is selected. And after re-adding the Long attribute it just simply context settings doing their job.

**Comment by janezd** [comment]
I understand why. But I'm not sure if it's desired. I suppose that `find_lat_lon` should return `None` if there is only a single useful variable. `find_lat_lon` does it's best to guess lat and long variables and guessing that it's one and the same variable is always wrong (except in a rare scenario in which all points lie on a "Earths diagonal" and one decided to save space by using a single variable :).

**Pull request merged**